### PR TITLE
Refactors analyze_activity_functions, pulling methods outside of the cron job. Also adds tests for some of the functions.

### PR DESCRIPTION
**Pull request by ashercodeorg**
This refactoring is in preparation to making significant changes to the analyze_day_fast function to correctly accommodate sampling factors. This PR will allow testing of the analyze_day_fast function in the followup PR.

Along the way, adding some quick tests of the hash methods.

### CODE REVIEW

**Comment by ashercodeorg** [comment]
@philbogle Is this good to merge into staging (after review)?

File: bin/test/analyze_hoc_activity_functions_test.rb [hunk 1]
```diff
@@ -0,0 +1,36 @@
+require 'minitest/autorun'
```
Comment (ashercodeorg) [inline bin/test/analyze_hoc_activity_functions_test.rb]: Is bin/test/ the appropriate directory for this (noting it is a new directory)?

**Comment by philbogle** [comment]
Looking

File: bin/analyze_hoc_activity_functions.rb [hunk 2]
```diff
@@ -0,0 +1,115 @@
+#!/usr/bin/env ruby
```
Comment (ashercodeorg) [inline bin/analyze_hoc_activity_functions.rb]: Is this the appropriate directory for these functions? Is this the appropriate filename? Should it be analyze_hoc_activity_helper.rb?

File: bin/test/analyze_hoc_activity_functions_test.rb (hunk 1 above)
Comment (aoby) [inline bin/test/analyze_hoc_activity_functions_test.rb]: Probably best to move the helper file into lib/ and the test into shared/test/

Comment (ashercodeorg) [inline bin/test/analyze_hoc_activity_functions_test.rb]: Done (both).

**Comment by aoby** [comment]
lgtm

**Comment by ashercodeorg** [comment]
@philbogle Good to submit to staging?

**Comment by philbogle** [comment]
(LGTM)

**Pull request merged**